### PR TITLE
[NEAT-424] 😎 Classic Importer

### PR DESCRIPTION
**Pull request by doctrino**
This is ready for review. It does not expose the new importer, so no addition to the changelog yet. 

It is not complete, that will be done later, after the end to end testing is done.

### CODE REVIEW

File: cognite/neat/constants.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
 DEFAULT_SPACE_URI = "http://purl.org/cognite/{space}#"
 DEFAULT_NAMESPACE = Namespace("http://purl.org/cognite/neat#")
+CLASSIC_CDF_NAMESPACE = Namespace("http://purl.org/cognite/cdf-classic#")
```
Comment (doctrino) [inline cognite/neat/constants.py:16]: Suggestion: Require this namespace in the ClassicImporter.

File: cognite/neat/graph/queries/_base.py [hunk 2]
```diff
@@ -95,6 +95,42 @@ def triples_of_type_instances(self, rdf_type: str) -> list[tuple[str, str, str]]
             )
             return []
 
+    def types_with_property(self, property_uri: URIRef) -> list[URIRef]:
```
Comment (doctrino) [inline cognite/neat/graph/queries/_base.py:98]: These are a sew helpers.

File: cognite/neat/rules/importers/_cdf_classic.py [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+        return ReadRules(rules, issues, read_context={})
+
+    def _add_source_system(self, rules: InformationInputRules) -> None:
+        types_ = self._store.queries.types_with_property(CLASSIC_CDF_NAMESPACE.source)
+        if not types_:
+            return
+        rules.classes.append(
+            InformationInputClass(
+                class_="SourceSystem",
+                parent="cdf_cdm:SourceSystem",
```
Comment (doctrino) [inline cognite/neat/rules/importers/_cdf_classic.py:48]: Thinking is that we create classes and use the `parent` or `reference` to link it to the core model.

File: cognite/neat/rules/importers/_cdf_classic.py [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+            ),
+        )
+        # TODO: Is this the correct syntax for the transformation?
+        transformation = [
+            f"classic_cdf:{remove_namespace_from_uri(type_)}(source) -> classic_cdf:SourceSystem(name)"
+            for type_ in types_
+        ]
```
Comment (doctrino) [inline cognite/neat/rules/importers/_cdf_classic.py:55]: Then we use the transformation to describe how the mapping needs to be done.

For example, `Event`, `Asset`, and `FileMetadata` has ta property source which is a string. This should be mapped to SourceSystem in the new core model.

File: cognite/neat/rules/importers/_cdf_classic.py [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+                parent=None,
+            ),
+        )
+        rules.properties.extend(
+            [
+                InformationInputProperty("DataSet", "name", "string"),
+                InformationInputProperty("DataSet", "description", "string"),
+                InformationInputProperty("DataSet", "external_id", "string"),
+                InformationInputProperty("DataSet", "write_protected", "boolean"),
+                InformationInputProperty("DataSet", "metadata", "json"),
```
Comment (doctrino) [inline cognite/neat/rules/importers/_cdf_classic.py:83]: We can consider to check whether there is any instance that has a metadata property for the Asset type. Then only add this property if there are instances that have it.

File: cognite/neat/rules/importers/_cdf_classic.py [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+                    "parent_id",
+                    "Asset",
+                    transformation="classic_cdf:Asset(parent_id) -> classic_cdf:Asset(parent)",
+                ),
+                InformationInputProperty("Asset", "description", "string"),
+                InformationInputProperty(
+                    "Asset",
+                    "dataSet",
+                    "DataSet",
+                    transformation="classic_cdf:Asset(dataSet_id) -> classic_cdf:Asset(dataSet)",
```
Comment (doctrino) [inline cognite/neat/rules/importers/_cdf_classic.py:115]: @nikokaoja Need some help to check if I remember this syntax correctly.

File: cognite/neat/rules/importers/_cdf_classic.py [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+                InformationInputProperty(
+                    "Asset",
+                    "tags",
+                    "string",
+                    max_count=10,
+                    transformation="classic_cdf:Asset(labels) -> classic_cdf:Asset(tags)",
+                ),
+            ]
+        )
+        # Todo Deal with GeoLocation
```
Comment (doctrino) [inline cognite/neat/rules/importers/_cdf_classic.py:128]: We also need to look up all relationships starting with an Asset, and figure out how it should be added.

File: tests/data/classic_windfarm.py [hunk 8]
```diff
@@ -0,0 +1,302 @@
+from cognite.client.data_classes import (
```
Comment (nikokaoja) [inline tests/data/classic_windfarm.py:1]: As you have put some thoughts into this classic dataset I am wandering do we want to maintain `./tests/data/class_cdf_extractore_data`, which I created with a similar idea (when testing classic cdf extractors) but I stored instances/resources as YAML files ?

**Review by nikokaoja — APPROVED** [review]

**Pull request merged**